### ISSUE TITLE
Cannot download some videos, possibly due to title

### ISSUE THREAD

**Issue by jnv** [zeronickname/VideoDownloader#1]
Hi, thanks for this awesome project!

I ran into issue with some videos from Strange Loop conference on YouTube, for example this one: ["I See What You Mean" by Peter Alvaro](https://www.youtube.com/watch?v=R2Aa4PivG0g). The download fails with <Untitled> Download unsuccessful. The same thing happens with for example this video: [Brian J Brennan: Being Human [JSConf2014]](https://www.youtube.com/watch?v=LlO2_GecWo8).
I can download other YouTube videos without any problem and the youtube-dl server returns a correct response with a correct video. I wonder whether the issue could be caused by the special characters, i.e. `["` in the title, so the download fails during the file creation?

**Comment by zeronickname** [comment]
You're almost certainly right as there's already a workaround to replace '#' charactes with spaces in the code. I couldn't find an exhaustive list of what characters I couldn't use in filenames in Android, but I can certainly add quotes and [ to the blacklist

**Comment by jnv** [comment]
Thanks for a quick response! I guess the filename limitations are driven by the file system; since SD cards are usually formatted as FAT32, its restrictions apply:

> Any byte except for values 0-31, 127 (DEL) and: " \* / : < > ? \ | + , . ; = [](lowcase a-z are stored as A-Z). ([source](http://stackoverflow.com/a/13502029)).

You could either blacklist these characters, like this ([source](http://stackoverflow.com/a/24692712)):

```
filename.replaceAll( "[\u0001-\u001f<>:\"/\\\\|?*\u007f]+", "" ).trim();
```

Or you could take a whitelist approach (e.g. just keep only alphanumeric characters, spaces and a few others), similar to [this](http://stackoverflow.com/a/35591188).

**Comment by zeronickname** [comment]
I don't think it's a FAT32 issue as the first time I hit this bug is with the '#' character which is allowed by FAT32 (and it's also not part of the suggested blacklist above). Regardless, I've swapped to a whitelist as it'll mean I don't have to keep revisiting this issue in future :)

Thanks for the suggestion. I've pushed the changes/precompiled apk to github and to the play store (it should roll out in the next few hours).

Let me know if it fixes it on your end too (once you get the chance to try it) and I'll close this issue
